picard/formats/id3.py

```diff
@@ -505,6 +507,8 @@ def _save(self, filename, metadata):
                             tags.add(self.build_TXXX(encoding, 'mood', values))
                     # No need to care about the TMOO tag being added again as it is
                     # automatically deleted by Mutagen if id2v23 is selected
+                        if frameid == 'TDRL':
```

Comment:
`TDRL` is specified in ID3v2.4, not  ID3v2.3.
So I guess we should only do that `if config.setting['write_id3v24']:`

Reply:
This is already part of the `if config.setting['write_id3v23']` block, same handling as `TMOO`. I think it is fine as it is.

Reply:
Do I miss something? But writing ID3v2.4 tag when `write_id3v23` doesn't look correct to me. What's the purpose of `write_id3v24` then?

Reply:
In case of v2.3 the code writes a TXXX tag(with `self.build_TXXX`) instead of the TDRL. The TDRL  itself will, if present, get removed by mutagen when downgrading to v2.3.